content/references/javascript-library/customizing-your-registration-schema/index.md

```diff
@@ -0,0 +1,123 @@
+---
+title: "Customizing Your Registration Schema"
+tags: ["LoginRadiusV2.js", "JavaScript Library", "Registration Schema", "Registration Fields"]
+description: "This is a document detailing registration schema customizations."
+path: "/references/javascript-library/customizing-your-registration-schema"
+---
+
+# Customizing Your Registration Schema
+
+Your application follows a configured schema which populates the registration forms generated using the LoginRadius JavaScript Library (LoginRadiusV2.js). You are free to override this default configuration during the initialization of the JavaScript Library.
+
+## Schema Format
+
+A schema object is defined using the following format:
+
+```js
+{
+  "type": "",
+  "name": "",
+  "display": "",
+  "rules": ""
+}
+```
+
+  * **type**: The type of field to display. This key supports `string` or `password` as possible values. `string` will display the consumer's input normally. `password` will hide the consumer's input.
+
+  * **name**: The name of the field to be added. This links the data entered in the field to a supported attribute in your app's user object. This key supports `firstname`, `lastname` and `company`.
+
+  * **display**: The text to display for your field in the registration form. You can enter any value here.
+
+  * **rules**: The validation rules to be used for this field. Refer to the [Validation Rules](#validation-rules) section for more information.
+
+## Adding Fields
+
+As mentioned in the JavaScript Library [Getting Started](/references/javascript-library/getting-started#registration) document, the registration schema objects are contained in the `LRObject.registrationFormSchema` array object.
+
+You can add to the default schema by utilizing the [Process Start Hook](/references/javascript-library/hooks#process-start-hook) and concatenating additional schema objects to the `LRObject.registrationFormSchema` array object.
+
+For example, here we are adding schemas utilizing the `company`, `firstname` and `lastname` data fields to `LRObject.registrationFormSchema`.
+
+```javascript
+  LRObject.$hooks.register("startProcess", function () {
+    LRObject.registrationFormSchema = LRObject.registrationFormSchema.concat([
```

Comment:
We can modify `LRObject.registrationFormSchema` before calling init for registration action.

But if you are including this inside the hook then you need to add below condition so that only at time of registration , formschema will update.
if(action === 'registration')

Reply:
I am modifying it inside the hook so that I can access the default fields as well instead of requiring the user to hardcode the values in as well. Added the condition.